### ISSUE TITLE
PullRequest Generator: unknown field "appSecretName" in io.argoproj.v1alpha1.ApplicationSet.spec.generators.pullRequest.github

### ISSUE THREAD

**Issue by Fettah** [argoproj/argo-cd#10910]
I am trying to connect Argocd to our github account and the PullRequest generators. 
after created the applicationSet with all the configs, kubectl was compalaining that the appSecretname is known 

Here is my config:
```
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: app-pr-testing
  namespace: argocd
spec:
  generators:
    - pullRequest:
        github:
          owner: my-company
          repo: my-repository
          labels:
            - argocd-deployment
         appSecretName: github-repo-creds
          .....
```

after a `kubectl apply -f` i get the following error:
`: error validating data: ValidationError(ApplicationSet.spec.generators[0].pullRequest.github): unknown field "appSecretName" in io.argoproj.v1alpha1.ApplicationSet.spec.generators.pullRequest.github; if you choose to ignore these errors, turn validation off with --validate=false`

ArgoCD Version: v2.412

**Comment by crenshaw-dev** [comment]
Ah, the `appSecretName` field won't be available until 2.5.

**Comment by crenshaw-dev** [comment]
Can you double-check that you deployed the latest CRD? I see that field in the CRD for 2.5.0-rc2: https://github.com/argoproj/argo-cd/blob/v2.5.0-rc2/manifests/crds/applicationset-crd.yaml#L2375

**Comment by Fettah** [comment]
I have noticed when you upgrade the helm chart, the CRDs do not get updated. so i still have the old CRD of the ApplicationSet in my kubernetes. even though i deleted argocd completey and reinstalled it it is still showing the old one
am trying now to delete the custom resource but unable to
<img width="1306" alt="image" src="https://user-images.githubusercontent.com/3440433/195563994-0a2762b9-57d0-4211-925b-cc41251955f5.png">

**Comment by Fettah** [comment]
I found the issue. the argo-helm that i use did not update their configs:
https://github.com/argoproj/argo-helm/blob/9920ecdb6e2126b41cc6b53c0bf41570e720e389/charts/argo-cd/templates/crds/crd-applicationset.yaml

The field does not show up there

**Comment by crenshaw-dev** [comment]
@Fettah I don't think argo-helm has released a chart for 2.5 yet. I'm sure it's planned, but I think they usually wait for the GA release.

**Comment by crenshaw-dev** [comment]
Closing since I believe this is an issue with an out of date CRD. Let me know if I'm incorrect!
